src/util/Cache.h

```diff
@@ -227,6 +226,44 @@ class FlexibleCache {
     // TODO<joka921>:: implement this functionality
   }
 
+  /// Return _maxSizeSingleEntry.
+  MemorySize getMaxSizeSingleEntry() const noexcept {
+    return _maxSizeSingleEntry;
+  }
+
+  /// Call `_valueSizeGetter` again for the value associated with the given key
+  /// and apply the diff to the internal size records. If the value grew too big
+  /// for the cache in the meantime it is removed, if it grew but fits in the
+  /// cache other values might be evicted instead. In case the value shrinks it
+  /// is also removed because then it will be incomplete so it no longer makes
+  /// sense to cache it.
```

Comment:
Clarify that if it suffices, then *other* entries are evicted, only if this wouldn't suffice, the current entry itself is erased.